transformations/slangificator/transformation.py

```diff
@@ -0,0 +1,271 @@
+import itertools
+import random
+import spacy
+import os
+
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+"""
+Base Class for implementing the different input transformations a generation should be robust against.
+"""
+
+
+def slangify(
+    self,
+    text,
+    probReplaceNoun=0.5,
+    probReplaceAdjective=0.5,
+    probReplaceAdverb=0.5,
+    seed=0,
+    max_outputs=1,
+):
+    random.seed(seed)
+
+    nlp = spacy.load("en_core_web_sm")  # get an instnace of the tikenizer
```

Comment:
minor typo: tokenizer

Reply:
Also, seems this is loading Spacy for every sentence and that can be costly. You might want to use spacy like [this](https://github.com/GEM-benchmark/NL-Augmenter/blob/d39f0e5f57a79937296351996617ac03ee80fb74/transformations/close_homophones_swap/transformation.py#L55).

Reply:
i.e. moving it to a constructor.

Reply:
Thanks for pointing to these issue @vyraun and @kaustubhdhole. Revised it in the updated code.